pkg/network/payload/notary_request.go

```diff
@@ -136,3 +136,20 @@ func (r *P2PNotaryRequest) isValid() error {
 	}
 	return nil
 }
+
+// Copy creates a deep copy of the P2PNotaryRequest.
+func (r *P2PNotaryRequest) Copy() *P2PNotaryRequest {
+	var mainTxCopy, fallbackTxCopy *transaction.Transaction
+	mainTxCopy = r.MainTransaction.Copy()
+
+	fallbackTxCopy = r.FallbackTransaction.Copy()
+
```

Comment:
A lot of useless empty lines. And `mainTxCopy`, `fallbackTxCopy` vars are not needed, initialize the `&P2PNotaryRequest` fields in-place.